docs/tutorials/gallery_adding_an_assistant.py

```diff
@@ -0,0 +1,177 @@
+"""
+# Adding an Assistant
+
+Ragna has builtin support for [several assistants][ragna.assistants], but there may be cases where you 
+want to use one that is not currently supported.
+
+This tutorial walks you through the basics of creating an assistant for LLMs that are not
+currently supported. 
+
+!!! note
+
+    This tutorial assumes that our TutorialAssistant class (shown below) is located in the
+    file ragna/assistants/_tutorial.py.
+
+
+"""
+
+# %%
+# ## The Finished Product
+
+# %%
+# !!! note
+#
+#     The code snippet below only includes up to Step 1. To actually include your source
+#     storage in Ragna, you must still perform Step 2.
+
+from typing import Iterator
+
+from ragna.core import Assistant, Source
+
+
+class TutorialAssistant(Assistant):
+    """
+    This is a basic assistant created for didactic purposes
+    """
+
+    def answer(self, prompt: str, sources: list[Source]) -> Iterator[str]:
+        """Answer a prompt given some sources.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
+        yield self._default_answer(prompt, sources)
+
+    def _default_answer(self, prompt: str, sources: list[Source]) -> str:
+        """
+        "Compute" the response to a given prompt. In this case,
+        the "computation" is a default string.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
+        return (
+            f"This is a default answer. There were {len(sources)} sources."
+            ""
+            f"The prompt was"
+            f"{prompt}"
+        )
+
+
+# %%
+# ## The Explanation
+
+# %%
+# !!! tip
+#
+#     For organizational purposes, this tutorial is divided into steps, but you may wish to perform
+#     Step 2 during Step 1 for manual testing or debugging.
+
+
+# %%
+# ### Step 0: Import Necessary Modules
+
+# %%
+# First start by importing some necessary components. Ragna uses the `typing` library:
+
+# %%
+# ```python
+# from typing import Iterator
+# ```
```

Comment:
This is not needed for two reasons:

1. The `Iterator` annotation is not necessary for this to function.
2. If one wishes to use type annotations, they will be familiar with the `typing` module. Since it is part of the standard library, it also comes pre-installed.